airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -361,7 +361,7 @@ private void putDoubleArray(final ObjectNode node, final String columnName, fina
     final ArrayNode arrayNode = Jsons.arrayNode();
     final ResultSet arrayResultSet = resultSet.getArray(colIndex).getResultSet();
     while (arrayResultSet.next()) {
-      arrayNode.add(DataTypeUtils.returnNullIfInvalid(() -> arrayResultSet.getDouble(colIndex), Double::isFinite));
+      arrayNode.add(DataTypeUtils.returnNullIfInvalid(() -> arrayResultSet.getDouble(2), Double::isFinite));
```

Comment:
What a find. Nice!

Reply:
Haha! I had been thinking about how this was working uptil now! We got lucky with `PostgresSourceDatatypeTest`. The `colIndex` value was 2 when we were processing the array. But with ctid addition the colIndex is no more 2 as one more column is added and thus the test starts failing!

Reply:
It pretty wild this bug managed to escape for so long because of index happened to be 2